### PR TITLE
CHAIN-582: Move to local validator, upgrade to v0.2.8, upgrade program/sdk files

### PR DESCRIPTION
**Pull request by bflood366**

### CODE REVIEW

File: arch/contracts/common/src/helper.rs [hunk 1]
```diff
@@ -410,29 +392,59 @@ pub fn get_processed_transaction(url: &str, tx_id: String) -> Result<ProcessedTr
...
+                    unreachable!("should not get here 1");
+                }
+            } else {
+                unreachable!("should not get here 2");
+            }
+        };
+
+        wait_time = 0;
+        while get_status(p.clone()) != "Processed".to_string()
+            && get_status(p.clone()) != "Failed".to_string()
```
Comment (bholzman) [inline arch/contracts/common/src/helper.rs:428]: nit: would possibly be cleaner to only call `get_status()` once

**Review by bholzman — APPROVED** [review]
Looks good

File: arch/contracts/common/src/helper.rs [hunk 2]
```diff
@@ -384,23 +381,8 @@ pub fn get_program(url: &str, program_id: String) -> String {
...
-        .expect("get_block should not fail")
-        .as_str()
-        .expect("cannot convert result to string")
-        .to_string()
-}
 
-/// Returns a processed transaction given the txid
-/// Keeps trying for a maximum of 60 seconds if the processed transaction is not available
-pub fn get_processed_transaction(url: &str, tx_id: String) -> Result<ProcessedTransaction> {
+pub fn get_processed_transaction<'a>(url: &str, tx_id: String) -> Result<ProcessedTransaction> {
```
[the remark this answers is not in the document]
Comment (bflood366) [reply arch/contracts/common/src/helper.rs]: removed

File: arch/contracts/common/src/helper.rs (hunk 1 above)
Comment (bflood366) [reply arch/contracts/common/src/helper.rs]: changed to call it once

**Pull request merged**